### ISSUE TITLE
Add a retry option to provisioners

### ISSUE THREAD

**Issue by AdamFoucher** [hashicorp/packer#8857]
Hello everyone,

while I was looking at Packer options for provisioners I didn't  find a retry option. I think it could be an interesting feature to add. This is what I had in mind: 
```
provisioners": [
    {
            "type": "powershell",
            "max_retries":3,
            "pause_before": "20s",
            "inline": [
               "Invoke-WebRequest -Uri <DownloadLink> -OutFile <FileFullPath>",
             ]
   }
]
```

In this example we download a file from an url and if the url is unreachable for some reason we retry 3 times and wait 20 seconds between each try.

I look forward to your answer!

**Comment by SwampDragons** [comment]
This is a nice idea. We do have a "retry" option when the --debug flag is set, but we don't have a way to access it for general runs, and it's a manual process. 

I think I'd be happy with adding a feature like this. That said, because it is fairly easy to write a shell wrapper to perform the retry for you, we probably won't prioritize it over issues that have no workarounds. If you or another community member are interested in trying to implement it before the internal team can get to it, I'd be happy to provide pointers and review.

**Comment by nywilken** [comment]
Hello :wave: this feature has been merged in #9061 and will be part of the v1.5.6 Packer release. If you would like to give it a try before the actual release it is available in the  [nightly Packer releases](https://github.com/hashicorp/packer/releases/tag/nightly).
